cmd/oras/internal/display/status/print.go

```diff
@@ -27,17 +27,33 @@ import (
 	"oras.land/oras-go/v2/registry"
 )
 
-var printLock sync.Mutex
+type Printer struct {
```

Comment:
IMHO this object should hold a writer, which accepts the output writer(out or stdout) of command. Printer should always prints to the writer.